flexget/plugins/modify/convert_magnet.py

```diff
@@ -33,8 +33,10 @@ def magnet_to_torrent(self, magnet_uri, destination_folder, timeout):
         import libtorrent
         params = libtorrent.parse_magnet_uri(magnet_uri)
         session = libtorrent.session()
-        # for some reason the info_hash needs to be bytes but it's a struct called sha1_hash
-        params['info_hash'] = bytes(params['info_hash'])
+        lt_version = [int(v) for v in libtorrent.version.split('.')]
+        if lt_version[0] > 0 or (lt_version[0] == 0 and lt_version[1] == 16 and lt_version[2] > 13):
```

Comment:
```
$ python2.7
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> magnet_uri='magnet:?xt=urn:btih:2fd26e37489b543179faefee0febf199b8f35698&dn=Indiana+Jones+and+the+Raiders+of+the+Lost+Ark+%281981%29+720p+BrRip+&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969&tr=udp%3A%2F%2Fzer0day.ch%3A1337&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969&tr=udp%3A%2F%2Fpublic.popcorn-tracker.org%3A6969'
>>> import libtorrent
>>> libtorrent.version
'1.0.7.0'
>>> params = libtorrent.parse_magnet_uri(magnet_uri)
>>> session = libtorrent.session()
>>> params['info_hash'] = bytes(params['info_hash'])
>>> handle = libtorrent.add_magnet_uri(session, magnet_uri, params)
>>> 

```
```
$ python3.5
Python 3.5.2 (default, Nov 17 2016, 17:05:23) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> magnet_uri='magnet:?xt=urn:btih:2fd26e37489b543179faefee0febf199b8f35698&dn=Indiana+Jones+and+the+Raiders+of+the+Lost+Ark+%281981%29+720p+BrRip+&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969&tr=udp%3A%2F%2Fzer0day.ch%3A1337&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969&tr=udp%3A%2F%2Fpublic.popcorn-tracker.org%3A6969'
>>> import libtorrent
>>> libtorrent.version
'1.0.7.0'
>>> params = libtorrent.parse_magnet_uri(magnet_uri)
>>> session = libtorrent.session()
>>> params['info_hash'] = bytes(params['info_hash'])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'sha1_hash' object is not iterable

```
```
$ python2.7
Python 2.7.6 (default, Oct 26 2016, 20:46:32) 
[GCC 4.8.4] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> magnet_uri='magnet:?xt=urn:btih:2fd26e37489b543179faefee0febf199b8f35698&dn=Indiana+Jones+and+the+Raiders+of+the+Lost+Ark+%281981%29+720p+BrRip+&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969&tr=udp%3A%2F%2Fzer0day.ch%3A1337&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969&tr=udp%3A%2F%2Fpublic.popcorn-tracker.org%3A6969'
>>> import libtorrent
>>> libtorrent.version
'0.16.13.0'
>>> params = libtorrent.parse_magnet_uri(magnet_uri)
>>> session = libtorrent.session()
>>> params['info_hash'] = bytes(params['info_hash'])
>>> handle = libtorrent.add_magnet_uri(session, magnet_uri, params)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: No registered converter was able to produce a C++ rvalue of type libtorrent::big_number from this Python object of type str
```
I don't have python3 on the machine that has libtorrent 0.16.13, so, I can't test it, sorry.

Seems that it is still broken (both before and after this patch) for python 3, for a different reason though.